src/sql/platform/serialization/common/serializationService.ts

```diff
@@ -35,41 +66,34 @@ export class SerializationService implements ISerializationService {
 
 	private disposables: IDisposable[] = [];
 
-	private _serverEvents: { [handle: number]: SerializationProviderEvents; } = Object.create(null);
-
-	private _lastHandle: number;
+	private providers: { providerId: string, provider: azdata.SerializationProvider }[] = [];
 
 	constructor(
 		@IConnectionManagementService private _connectionService: IConnectionManagementService,
 		@ICapabilitiesService private _capabilitiesService: ICapabilitiesService
 	) {
 	}
 
-	public addEventListener(handle: number, events: SerializationProviderEvents): IDisposable {
-		this._lastHandle = handle;
-
-		this._serverEvents[handle] = events;
+	public registerProvider(providerId: string, provider: azdata.SerializationProvider): void {
+		this.providers.push({ providerId: providerId, provider: provider });
+	}
 
-		return {
-			dispose: () => {
-			}
-		};
+	public hasProvider(): boolean {
+		return this.providers.length > 0;
 	}
 
-	public saveAs(saveFormat: string, savePath: string, results: string, appendToFile: boolean): Thenable<azdata.SaveResultRequestResult> {
-		if (this._serverEvents === undefined || this._serverEvents[this._lastHandle] === undefined) {
-			return this.disabledSaveAs();
-		}
 
-		return this._serverEvents[this._lastHandle].onSaveAs(saveFormat, savePath, results, appendToFile);
```

Comment:
FYI - this never worked previously. I don't think we lose anything by replacing with a promise rejection, and in a follow up PR I want to plumb through full support here instead, but that'll require some additional API changes.